rmw_zenoh_cpp/src/detail/zenoh_router_check.hpp

```diff
@@ -0,0 +1,27 @@
+// Copyright 2023 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DETAIL__ZENOH_ROUTER_CHECK_HPP_
+#define DETAIL__ZENOH_ROUTER_CHECK_HPP_
+
+#include <zenoh.h>
+
+#include "rmw/ret_types.h"
+
+/// Check if a Zenoh router is connected to the session.
+/// @param session Zenoh session to check.
+/// @return RMW_RET_OK if a Zenoh router is connected to the session.
+rmw_ret_t zenoh_router_check(z_session_t session);
```

Comment:
Yeah it is true we are not using this more than once. The only reason to have a separate file with this method is to make the `rmw_init.cpp` file a bit more clean and if at some we want to add tests this will be useful.
(Without mention that if we want to launch the zenoh router from here probably another method might be implementeed)

Instead of having just a sole method in this header file we can merge this file with the `zenoh_config.hpp` header and create a `zenoh_utils.hpp` file.

If you still recommend having it all in the same `rmw_init.cpp` file then I am ok with moving this code into the `rmw_init.cpp`. ptal @Yadunund

Reply:
It's a little overkill to have this header for just one function, but I'm fine with this for now.  If we never end up adding anything here, we can always combine this into `rmw_init` later on.